orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/WaitForManifestStableContext.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 Cerner Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.manifest;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.Getter;
+
+@Getter
+public class WaitForManifestStableContext extends HashMap<String, Object> {
+
+  private final Optional<List<String>> messages;
+  private final Optional<Exception> exception;
+  private final Optional<List<Map<String, String>>> stableManifests;
+  private final Optional<List<Map<String, String>>> failedManifests;
+  private final Optional<List> warnings;
+
+  // There does not seem to be a way to auto-generate a constructor using our current version of
+  // Lombok (1.16.20) that
+  // Jackson can use to deserialize.
+  public WaitForManifestStableContext(
+      @JsonProperty("messages") Optional<List<String>> messages,
```

Comment:
Went the route of using Optionals for the getters here since it makes it nice to either get the existing context field or initialize to a new array list in the main task logic using orElseGet.

Reply:
In retrospect I can also just do that logic in this constructor - set to the deserialized value if non-null or initialize a new array list otherwise). Really no reason to do it in the task logic. Thoughts?

Reply:
Yeah, I wonder if just having the constructor do `Optional.ofNullable(messages).orElse(...)` would be a bit cleaner---but up to you, I think this is fine too.  (I didn't realize that Jackson had support for using `Optional` in this way, but it must as your test case is exercising this.)

Reply:
Alright, made the context cleaner to consume. In the context I still make use of the Jackson Optional support though, since it essentially does the `Optional.ofNullable(...)` part for me.